core/trino-main/src/main/java/io/trino/server/TrinoSystemRequirements.java

```diff
@@ -129,6 +130,13 @@ private static void verifyUsingG1Gc()
         }
     }
 
+    private static void verifyVectorApiEnabled()
+    {
+        if (ModuleLayer.boot().findModule("jdk.incubator.vector").isEmpty()) {
+            failRequirement("Trino requires the JDK vector API to be enabled");
```

Comment:
I'd phrase it as:

`Trino requires Vector API to be enabled/linked at runtime`